application/models/Question.php

```diff
@@ -529,16 +529,13 @@ public function deleteAllAnswers(array $exceptIds = [])
         $ids = array_merge([$this->qid], $this->allSubQuestionIds);
         $qidsCriteria = (new CDbCriteria())->addInCondition('qid', $ids);
         $qidsCriteria->addNotInCondition('aid', $exceptIds);
-        $answerIds = [];
         $answers = Answer::model()->findAll($qidsCriteria);
         if (!empty($answers)) {
             foreach ($answers as $answer) {
-                $answerIds[] = $answer->aid;
+                AnswerL10n::model()->deleteAllByAttributes(['aid' => $answer->aid]);
```

Comment:
Must delete AnswerL10n only after answer was deleted with success. 

If we have an Answer without AnswerL10n : this broke when try to see survey.

Reply:
I agree